emu/src/cpu/arm7tdmi.rs

```diff
@@ -770,6 +772,42 @@ impl Arm7tdmi {
             _ => unreachable!(),
         }
     }
+
+    pub(crate) fn push_pop_register(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let load_store: LoadStoreKind = op_code.get_bit(11).into();
+        let store_or_not_reg_lr = op_code.get_bit(8);
+        let rlist = op_code.get_bits(0..=7);
+
+        match load_store {
+            LoadStoreKind::Store if !store_or_not_reg_lr => todo!(),
+            LoadStoreKind::Store if store_or_not_reg_lr => {
+                let reg_sp = self.registers.register_at(REG_SP);
+                self.registers.set_register_at(REG_SP, reg_sp - 4);
+                let stack_addr = (self.registers.register_at(REG_SP) & !3) as usize;
+                self.memory
+                    .lock()
+                    .unwrap()
+                    .write_word(stack_addr, self.registers.register_at(REG_LR));
```

Comment:
Also instead of splitting it in two match arms, what do you whink if we do the if inside the `LoadStoreKind::Store` arm?

Something like that:

```rust
LoadStoreKind::Store => {
    if store_or_not_reg_lr {
        let reg_sp = self.registers.register_at(REG_SP);
        self.registers.set_register_at(REG_SP, reg_sp - 4);
        let stack_addr = (self.registers.register_at(REG_SP) & !3) as usize;
        self.memory
            .lock()
            .unwrap()
            .write_word(stack_addr, self.registers.register_at(REG_LR));
    }
    ...
}, ...
```

Reply:
Improved